base_layer/wallet_ffi/src/lib.rs

```diff
@@ -180,20 +184,33 @@ pub unsafe extern "C" fn string_destroy(ptr: *mut c_char) {
 /// ## Arguments
 /// `byte_array` - The pointer to the byte array
 /// `element_count` - The number of elements in byte_array
+/// `error_out` - Pointer to an int which will be modified to an error code should one occur, may not be null. Functions
```

Comment:
Remind me: Does the client have to do anything to prepare the memory this points to for this to work? If so then explain it in the comment.

Reply:
The client doesn't have to do anything complicated here, simple as:
```
int var = 0; //or any valid int  
int *ptr;  //declare pointer
ptr = &var; //assign address of var to pointer
//pass ptr to function
//check value of var on return if needed
```